### PR TITLE
Achievements saving

### PR DESCRIPTION
**Pull request by jessearens**
With #219 merged, the failing builds are now solved and therefore this can be reviewed again.

<feature

### CODE REVIEW

File: src/main/java/com/github/fishio/achievements/AchievementIO.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import com.esotericsoftware.yamlbeans.YamlWriter;
+import com.github.fishio.logging.Log;
+import com.github.fishio.logging.LogLevel;
+
+/**
+ * This class is responsible for the Input and Output of Achievements to and
+ * from file, making sure that the achievements are saved and loaded.
+ *
+ */
+public class AchievementIO {
```
Comment (ChielBruin) [inline src/main/java/com/github/fishio/achievements/AchievementIO.java]: _Auto-generated comment_

```
Utility classes should not have a public or default constructor.
```

Just add 'private AchievementIO() {}'
<layout

**Comment by ChielBruin** [comment]
Looks good! 
You only have to fix the little checkstyle error.

<feature

File: src/main/java/com/github/fishio/achievements/AchievementIO.java (hunk 1 above)
Comment (jessearens) [inline src/main/java/com/github/fishio/achievements/AchievementIO.java]: Well spotted! I had already seen this but forgotten to fix it.
Just making a private constructor causes another checkstyle complaint, so I also made the class final with 59f2e6b

<layout,logic

File: src/test/java/com/github/fishio/achievements/TestAchieveEnemyKill.java [hunk 2]
```diff
@@ -37,7 +37,10 @@ public EnemyKillObserver getKillObserver() {
 		
 		when(sppf.getPlayer()).thenReturn(pf);
 		when(sppf.playerProperty()).thenReturn(sop);
-		return new EnemyKillObserver(sppf);
+		EnemyKillObserver eko = new EnemyKillObserver(sppf);
+		eko.attachAchievement(AchievementManager.ENEMY_KILL);
```
Comment (Taeir) [inline src/test/java/com/github/fishio/achievements/TestAchieveEnemyKill.java:41]: `eko.attachAchievement(AchievementManager.ENEMY_KILL);` can be removed, it already happens in the constructor of EnemyKillObserver.

<logic

**Comment by Taeir** [comment]
Apart from that one minor thing I mentioned, this looks good and can be merged.

<review

**Pull request merged**